unpackaged/vesuvio_calibration/calibrate_vesuvio_uranium_martyn_MK5.py

```diff
@@ -668,81 +668,88 @@ def _fit_peaks(self):
+
+  def _fit_peak(self, peak_index, spec_index, peak_position, output_parameter_table_name, output_parameter_table_headers):
+    spec_number = self._spec_list[0] + spec_index
+    self._prog_reporter.report("Fitting peak %d to spectrum %d" % (peak_index, spec_number))
+
+    peak_params = self._find_peaks_and_output_params(peak_index, spec_number, spec_index, peak_position)
+    fit_func_string = self._build_function_string(peak_params)
+    xmin, xmax = self._find_fit_x_window(peak_params)
+    fit_output_name = '__' + self._output_workspace_name + '_Peak_%d_Spec_%d' % (peak_index, spec_index)
+    status, chi2, ncm, fit_params, fws, func, cost_func = Fit(Function=fit_func_string, InputWorkspace=self._sample,
+                                                              IgnoreInvalidData=True,
+                                                              StartX=xmin, EndX=xmax, WorkspaceIndex=spec_index,
                                                               CalcErrors=True, Output=fit_output_name,
                                                               Minimizer='Levenberg-Marquardt,RelError=1e-8')
 
-        #output fit parameters to table workspace
-        fit_values = dict(zip(params.column(0), params.column(1)))
-        fit_errors = dict(zip(params.column(0), params.column(2)))
+    self._output_fit_params_to_table_ws(spec_number, fit_params, output_parameter_table_name,
+                                        output_parameter_table_headers)
+    fit_workspace_name = fws.name()
+    self._del_fit_workspaces(ncm, fit_params, fws)
 
-        row_values = [fit_values[name] for name in param_names]
-        row_errors = [fit_errors[name] for name in param_names]
-        row = [element for tupl in zip(row_values, row_errors) for element in tupl]
+    return fit_workspace_name
 
-        mtd[peaks_table].addRow([spec_number] + row)
+  def _find_peaks_and_output_params(self, peak_index, spec_number, spec_index, peak_position):
+    peak_table_name = '__' + self._sample + '_peaks_table_%d_%d' % (peak_index, spec_index)
+    find_peak_params = self._get_find_peak_parameters(spec_number, [peak_position])
+    FindPeaks(InputWorkspace=self._sample, WorkspaceIndex=spec_index, PeaksList=peak_table_name, **find_peak_params)
+    if mtd[peak_table_name].rowCount() == 0:
+      logger.error('FindPeaks could not find any peaks matching the parameters:\n' + str(find_peak_params))
+      sys.exit()
 
-        self._parameter_tables.append(peaks_table)
-        self._peak_fit_workspaces.append(fws.name())
+    peak_params = mtd[peak_table_name].row(0)
+    DeleteWorkspace(peak_table_name)
+    return peak_params
 
-        DeleteWorkspace(ncm)
-        DeleteWorkspace(params)
-        if not self._create_output:
-          DeleteWorkspace(fws)
+  def _find_fit_x_window(self, peak_params):
+    xmin, xmax = None, None
 
-      self._fit_workspaces.append(self._peak_fit_workspaces)
+    position = '' + self._func_param_names['Position']
```

Comment:
This is just a preference. I think it is clearer to use `str`
```suggestion
    position =  str(self._func_param_names['Position'])
```

Reply:
Agreed, I am planning to submit a PR after this batch is merged which will update the whole script with small best practice changes like this.